### ISSUE TITLE
hwtest: Instead of replacer, run a binary archive?

### ISSUE THREAD

**Issue by geohot** [dougallj/applegpu#30]
The wrap/replacer.dylib API is unstable. Any reason you can't load a custom shader from a `BinaryArchive` and run it? Do you need something lower level for the hardware tests?

**Comment by dougallj** [comment]
Yeah, that's a good idea – I just wasn't aware of MTLBinaryArchive at the time. I haven't tried overwriting code in binary archives, but assuming that works it should be a much more stable strategy.
